RecoTracker/DeDx/plugins/DeDxHitCalibrator.cc

```diff
@@ -0,0 +1,392 @@
+  dedxCalib_ = iSetup.getHandle(dedxCalibToken_);
+  tkGeom_ = iSetup.getHandle(tkGeomToken_);
+}
+
+void DeDxHitCalibrator::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  const auto& tracks = iEvent.getHandle(tracksToken_);
+  const auto& dedxHitInfo = iEvent.get(dedxHitInfoToken_);
+  pixelCalib_.setESObjects(iSetup);
+
+  // creates the output collection
+  auto pixelTrackDeDxHitAss = std::make_unique<reco::TrackDeDxHitsCollection>(reco::TrackRefProd(tracks));
+  auto stripTrackDeDxHitAss = std::make_unique<reco::TrackDeDxHitsCollection>(reco::TrackRefProd(tracks));
+
+  for (size_t i = 0; i < tracks->size(); i++) {
+    const auto& track = reco::TrackRef(tracks, i);
+    const auto& dedxHits = dedxHitInfo[track];
+    if (dedxHits.isNull())
+      continue;
+    reco::DeDxHitCollection pixelHits, stripHits;
+    processHitInfo(*dedxHits, track->p(), pixelHits, stripHits);
+    pixelTrackDeDxHitAss->setValue(i, pixelHits);
+    stripTrackDeDxHitAss->setValue(i, stripHits);
+  }
+
+  iEvent.put(std::move(pixelTrackDeDxHitAss), "PixelHits");
+  iEvent.put(std::move(stripTrackDeDxHitAss), "StripHits");
+}
+
+/*****************************************************************************/
+void DeDxHitCalibrator::processHitInfo(const reco::DeDxHitInfo& info,
+                                       const float& trackMomentum,
+                                       reco::DeDxHitCollection& pixelHits,
+                                       reco::DeDxHitCollection& stripHits) {
+  for (size_t i = 0; i < info.size(); i++) {
+    // Require hits to be complete and compatible
+    const auto& type = info.type(i);
+    if (!(type & (1 << reco::DeDxHitInfo::Complete)) || !(type & (1 << reco::DeDxHitInfo::Compatible)))
+      continue;
+
+    // Effective path length
+    const auto& pl = info.pathlength(i);
+    const auto pathLength = pl * (1. + 0.07 * std::log(pl / 450e-4));
+    const DetId& detId = info.detId(i);
+
+    // Strip
+    if (const auto& stripCluster = info.stripCluster(i)) {
+      const auto& thickness =
+          dynamic_cast<const StripGeomDetUnit*>(tkGeom_->idToDet(detId))->surface().bounds().thickness();
+      const auto& det = getDetId(detId, thickness);
+      const auto& chipId = ChipId(detId, stripCluster->barycenter() / 128.);
+      // Fill measured strip deposits
+      const auto& thr = dedxCalib_->thr[det];
+      std::vector<std::pair<double, int> > b;
+      b.reserve(stripCluster->amplitudes().size() + 2);
+      b.emplace_back(thr, isBelow);
+      for (const auto& adc : stripCluster->amplitudes()) {
+        if (adc > 253)
+          b.emplace_back(254., isOver);
+        else
+          b.emplace_back(adc + 0.5, isNormal);
+      }
+      b.emplace_back(thr, isBelow);
+      // Fit
+      const auto& result = fitStripCluster(b, dedxCalib_->alpha[det], 1. / dedxCalib_->sigma[det]);
+      // ADC -> e- -> MeV
+      const auto& energy = correctEnergy(result.first * 217 * 3.61e-6, chipId);
```

Comment:
Changed to MeVPerElectron parameter to clarify the number in d5037052739eade4f196f6779f9c753c25ba9784

Reply:
> Changed to MeVPerElectron parameter to clarify the number in https://github.com/cms-sw/cmssw/commit/d5037052739eade4f196f6779f9c753c25ba9784

Thanks for this, can the input value be passed explicitly in the corresponding python configuration (that at the moment is only auto-generated) and such that is copied by value from https://github.com/cms-sw/cmssw/blob/master/SimGeneral/MixingModule/python/SiStripSimParameters_cfi.py#L19 (so that the two stay always in synch)? 
Something like:

https://github.com/cms-sw/cmssw/blob/356b35d2da8911a64972d908f96f2e7244d0c2d1/CalibTracker/SiPixelLorentzAngle/python/SiPixelLorentzAnglePCLHarvesterMCS_cfi.py#L2-L7

Reply:
Addressed in a60fd6335930bef2c684dbe968757a93e5ab9f93